cg/meta/deliver_ticket.py

```diff
@@ -135,48 +88,15 @@ def concatenate(self, ticket: str, dry_run: bool) -> None:
                 continue
             if not dir_path.is_dir():
                 continue
-            for read_direction in [1, 2]:
-                same_direction: list[Path] = self.get_current_read_direction(
-                    dir_path=dir_path, read_direction=read_direction
-                )
-                total_size: int = self.get_total_size(files=same_direction)
-                output: Path = self.generate_output_filename(
-                    date=date, dir_path=dir_path, read_direction=read_direction
-                )
-                if dry_run:
-                    for file in same_direction:
-                        LOG.info(f"Dry run activated, {file} will not be appended to {output}")
-                else:
-                    LOG.info(f"Concatenating sample: {dir_path.name}")
-                    self.concatenate_same_read_direction(reads=same_direction, output=output)
-                if dry_run:
-                    continue
-                concatenated_size = output.stat().st_size
-                if total_size != concatenated_size:
-                    raise CgError("WARNING data lost in concatenation")
-
-                LOG.info(
-                    "QC PASSED: Total size for files used in concatenation match the size of the concatenated file"
-                )
-                self.remove_files(reads=same_direction)
-
-    def get_app_tag(self, samples: list) -> str:
-        app_tag = samples[0].application_version.application.tag
-        return app_tag
-
-    def check_if_concatenation_is_needed(self, ticket: str) -> bool:
-        cases: list[Case] = self.get_all_cases_from_ticket(ticket=ticket)
-        case_id = cases[0].internal_id
-        case_obj = self.status_db.get_case_by_internal_id(internal_id=case_id)
-        samples: list[Sample] = [link.sample for link in case_obj.links]
-        app_tag = self.get_app_tag(samples=samples)
-        for prefix in PREFIX_TO_CONCATENATE:
-            if app_tag.startswith(prefix):
-                LOG.info(
-                    f"Identified {app_tag} as application tag, i.e. the fastqs should be concatenated",
-                )
-                return True
-        LOG.info(
-            f"The following application tag was identified: {app_tag}, concatenation will be skipped",
-        )
-        return False
+            forward_out_path = self.fastq_file_service.get_concatenated_forward_read_output_path(
```

Comment:
I think this design can be improved. The `get_concatenated_forward_read_output_path` and `get_concatenated_reverse_read_output_path` should not be a part of the `FastqFileService. 

They are internal implementation details specific to this workflow and makes the `FastqFileService` interface less clean.

Reply:
I think I followed what you and Vadym envisioned in the next push, resolving this for now!

Reply:
If you want, I can refactor it later? It is jpoor design to couple the microbial naming requirements to the implementation of a general `FastqFileService`

Reply:
I have an idea for it, no need 👍